tests/linux_kernel/kmod/drgn_test.c

```diff
@@ -739,6 +740,68 @@ static void drgn_test_xarray_exit(void)
 #endif
 }
 
+// idr
+
+DEFINE_IDR(drgn_test_idr_empty);
+DEFINE_IDR(drgn_test_idr_one);
+DEFINE_IDR(drgn_test_idr_one_at_zero);
+DEFINE_IDR(drgn_test_idr_sparse);
+DEFINE_IDR(drgn_test_idr_dense);
+static unsigned long drgn_test_idr_ptrs[5] = {0x12121212, 0x34343434,
+					      0x5656565656565656,
+				              0x1234567812345678,
+					      0x1234567890abcdef};
+static int drgn_test_idr_init(void)
+{
+	int ret, i;
+
+	ret = idr_alloc(&drgn_test_idr_one, (void *)0xdeadb00, 66, 67,
+				GFP_KERNEL);
+	if (ret != 66)
+		return -ENOSPC;
```

Comment:
In the kernel, it seems like the usual error checking pattern for `idr_alloc()` is:

```suggestion
	if (ret < 0)
		return ret;
```

Reply:
Yeah. The thing I had in mind was that since I am allocating a single ID from a range consisting of 1 number only, I should explicitly check if I got that number as returned ID. But this should not be needed since idr allocation code would take care of range consisting of one number and would return error if it can't allocate that ID. So just checking "ret" will be enough